docs/source/gs-processing/usage/emr-serverless.rst

```diff
@@ -0,0 +1,273 @@
+* An EMR-S application that uses our custom image.
+* An execution role that our EMR-S jobs will use when we launch them.
+
+To launch the same example job as we demonstrate in the :doc:`SageMaker Processing job guide <amazon-sagemaker>`
+you can use the following ``bash`` snippet. Note that we use ``jq`` to wrangle JSON data,
+which you can can from its `official website <https://jqlang.github.io/jq/download/>`_,
+using your package manager, or by running ``pip install jq``.
+
+Before starting  the job, make sure you have uploaded the input data
+as described in :ref:`gsp-upload-data-ref`.
+
+.. code-block:: bash
+
+    APPLICATION_ID="enter-your-application-id-here"
+    ACCOUNT=$(aws sts get-caller-identity --query Account --output text)
+    MY_BUCKET="enter-your-bucket-name-here"
+    EMR_S_ROLE_NAME="enter-your-emr-serverless-execution-role-name-here"
+    REGION="bucket-region" # e.g. us-west-2
+    INPUT_PREFIX="s3://${MY_BUCKET}/gsprocessing-input"
+    OUTPUT_BUCKET=${MY_BUCKET}
+    GRAPH_NAME="small-graph"
+    CONFIG_FILE="gconstruct-config.json"
+    NUM_FILES="4"
+    GSP_HOME="enter/path/to/graphstorm/graphstorm-processing/"
+
+    LOCAL_ENTRY_POINT=$GSP_HOME/graphstorm_processing/distributed_executor.py
+    S3_ENTRY_POINT="s3://${OUTPUT_BUCKET}/emr-serverless-scripts/distributed_executor.py"
+
+    ROLE="arn:aws:iam::${ACCOUNT}:role/${EMR_S_ROLE_NAME}"
+
+
+    export OUTPUT_PREFIX="s3://${OUTPUT_BUCKET}/gsprocessing/emr-s/${GRAPH_NAME}/${NUM_FILES}files/"
+
+    # Copy entry point script to S3 to ensure latest version is used
+    aws s3 cp $LOCAL_ENTRY_POINT $S3_ENTRY_POINT
+
+    # Construct arguments JSON string using jq
+    ARGS_JSON=$( jq -n \
+        --arg entry "$S3_ENTRY_POINT" \
+        --arg in "$INPUT_PREFIX" \
+        --arg out "$OUTPUT_PREFIX" \
+        --arg cfg "$CONFIG_FILE" \
+        --arg nfiles "$NUM_FILES" \
+        --arg gname "$GRAPH_NAME" \
+        '{
+            sparkSubmit: {
+                entryPoint: $entry,
+                entryPointArguments:
+                    ["--input-prefix", $in,
+                    "--output-prefix", $out,
+                    "--config-file", $cfg,
+                    "--num-output-files", $nfiles,
+                    "--graph-name", $gname]
+            }
+        }' )
+
+    echo "Arguments JSON:"
+    echo $ARGS_JSON | jq -r
+
+    echo "Starting EMR-S job..."
+    aws --region $REGION emr-serverless start-job-run \
+        --name "gsprocessing-emr-s-example" \
+        --application-id $APPLICATION_ID \
+        --execution-role-arn $ROLE \
+        --job-driver "${ARGS_JSON}" # Need to surround ARGS_JSON with quotes here to maintain JSON formatting
+
+As with the SageMaker example, we need to run a follow-up job to align the output with the
+expectations of the DistDGL partitioning pipeline. The easiest is to run the job locally
+on an instance with S3 access:
+
+.. code-block:: bash
```

Comment:
Shall we have a standalone chapter talking about partition since both SageMaker and EMR-S will use the same steps to do graph partition and dispatch.

Reply:
Probably would make sense. Currently we cover DistPart in the GSF-on-SageMaker documentation I think, think we should move it under the GSProcessing documentation? The thing is that DistPart requires the GSF image to be executed and not the GSProcessing one, so it might make more sense to be under GSF docs.

Reply:
@zhjwy9343